File: Programmers/유석환/Q42587.swift
import Foundation

func solution(_ priorities:[Int], _ location:Int) -> Int {
    var arr = priorities
    var loc = location
    var cnt = 0
    
    while true {
        if loc == 0 {
            if arr[0] == arr.max() {
                cnt += 1
                break
            } else {
                arr.append(arr[0])
                arr.remove(at: 0)
                loc = arr.count - 1
            }
        } else {
            if arr[0] == arr.max() {
                cnt += 1
                arr.remove(at: 0)
            } else {
                arr.append(arr[0])
                arr.remove(at: 0)
            }
            loc -= 1
        }
    }
    
    return cnt
}

File: Programmers/노수일/Q42587.py
from collections import deque


def solution(priorities, location):
    answer = 0
    print_queue = deque([(value, idx) for idx, value in enumerate(priorities)])

    while True:
        next_print = print_queue.popleft()
        if print_queue and max(print_queue)[0] > next_print[0]:
            print_queue.append(next_print)
        else:
            answer += 1
            if next_print[1] == location:
                break

    return answer
